src/amqp.js

```diff
@@ -0,0 +1,132 @@
+var amqplib = require('amqplib/callback_api');
+var debug = require('debug')('feathers-sync');
+
+var ch;
+var conf ={};
+var listSubscriber = [];
+
+var exchangeName = 'feathers-sync';
+
+var onmessage = function(ev,data){
+  if(listSubscriber[ev] && Array.isArray(listSubscriber[ev])){
+    listSubscriber[ev].forEach(function(cb){
+      cb(data);
+    });
+  }
+
+};
+
+var connection = function(cb){
+
+  amqplib.connect(conf.uri, conf.amqpConnectionOptions, function (err, conn){
+    if(err){
+      debug('Error while connecting to AMQP: ', err.toString());
+    }else{
+      ch = conn.createChannel();
+
+      ch.assertExchange(exchangeName, 'fanout', {durable: false});
+      ch.assertQueue('', {exclusive: true}, function(err, q) {
+        if(err){
+          debug('Error while assertQueue to AMQP: ', err.toString());
+        }else{
+          ch.bindQueue(q.queue, exchangeName, '');
+           if(typeof cb === 'function'){
+             cb(ch);
+           }
+          ch.consume(q.queue, function(msg) {
+            var temp = JSON.parse(msg.content.toString());
+            onmessage(temp.ev,temp.data);
+          }, {noAck: true});
+        }
+      });
+    }
+  });
+};
+
+var subscribe = function(ev,cb){
+  if(listSubscriber[ev]){
+    listSubscriber[ev].push(cb);
+  }else{
+    listSubscriber[ev] = [cb];
+  }
+};
+
+var publish = function(ev,data){
+  var msg = {};
+  msg.ev = ev;
+  msg.data = data;
+  //amqp name
+  if(ch){
```

Comment:
I'd like to see these variables longer. I know that `ch` means `channel` but people new to RabbitMQ may not.